sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -44,6 +44,9 @@ object UnsupportedOperationChecker extends Logging {
         throwError("dropDuplicatesWithinWatermark is not supported with batch " +
           "DataFrames/DataSets")(d)
 
+      case t: TransformWithStateInPandas =>
```

Comment:
Let's file a JIRA ticket with TODO like this.

```
        // TODO(SPARK-40443): support applyInPandasWithState in batch query
        throw new SparkUnsupportedOperationException("_LEGACY_ERROR_TEMP_3176")
```

I see my inconsistency here - the above is in BasicOperators in SparkStrategies. 

https://github.com/apache/spark/blob/f7122137006e941393c8be619fb51b3b713a24cb/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala#L966-L968

Shall we move this to BasicOperators with adding TODO comment like the above? Let's leave this place to be something there is a reason not to support and won't support in future.